docs/modules/reference/pages/service-assurance/monitors/PtpMonitor.adoc

```diff
@@ -0,0 +1,60 @@
+
+= PtpMonitor
+:description: Learn how to configure and use the PtpMonitor in OpenNMS {page-component-title} to monitor the correct PTP port state for an interface.
+
+The Precision Time Protocol allows the precise synchronization of clocks in measurement and control systems implemented with packet-based networks.
+OpenNMS can monitor the port state of SNMP-enabled interfaces by using the PTP MIB defined in RFC 8173.
+During the poll, it queries the device for the available clock-ports and their states and compares with a given expected state.
+If the state equals the desired state the service is considered up and down otherwise.
+
+== Monitor facts
+
+[cols="1,7"]
+|===
+| Class Name
+| `org.opennms.netmgt.poller.monitors.PtpMonitor`
+|===
+
+== Configuration and Use
+
+.Optional monitor-specific parameters for the PtpMonitor
+[options="header"]
+[cols="1,3,2"]
+|===
+| Parameter
+| Description
+| Default
+
+| clock-port
+| Name of the interface that must be in the specified state.
+| n/a
+
+| port-state
+| The desired state that the port must be in.
+| n/a
+
+|===
+
+This monitor implements the <<reference:service-assurance/introduction.adoc#ref-service-assurance-monitors-common-parameters, Common Configuration Parameters>>.
+
+== Examples
+
+Example uses CentOS/RHEL path name.
+For Debian/Ubuntu, use `/var/lib/opennms/rrd/response`.
+
+Note that you must include the `monitor` section for each service in your definition.
+
+[source, xml]
+----
+ <service name="PTP" interval="300000" user-defined="false" status="on">
+     <pattern><![CDATA[^PTP-(?<clockPort>.*)-(?<portState>.*)$]]></pattern>
```

Comment:
Just a personal opinion, I would prefer to let people use MetaData key:value pairs as configuration parameters for the monitor instead of deriving it from the service monitor name. It overloads the service name and can lead to a lot of problems. Renaming a service has side effects that aren't transparent and can break things. MetaData on the service is a much cleaner way to deal with service-specific configurations. Using a pattern match to apply the service configuration to a given set of service names would make sense to me.

Reply:
Yes I know, in my opinion suggesting all possibilities on how you can inject configuration parameters is a bit harmful in this example. People copy and paste this as is and cause more trouble to start with. I suggest making a pattern match just for the service name and removing the regexp groups deriving service config parameters from the service name. Let that be an advanced option they have to pick when users really know what they are doing.